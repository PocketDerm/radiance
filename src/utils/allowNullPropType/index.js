export default wrapper => (props, propName, ...rest) =>
  props[propName] === null ? null : wrapper(props, propName, ...rest);