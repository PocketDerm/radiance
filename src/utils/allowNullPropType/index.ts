export default (wrapper: any) => (
  props: any,
  propName: string,
  ...rest: any[]
) => (props[propName] === null ? null : wrapper(props, propName, ...rest));
