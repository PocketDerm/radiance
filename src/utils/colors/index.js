import tinycolor from 'tinycolor2';

function percentToInt(str) {
  // '20%' -> 20
  return parseInt(str.replace(/%/g, ''), 10);
}

export function lighten(color, modifier) {
  return tinycolor(color)
    .lighten(percentToInt(modifier))
    .toString();
}

export function darken(color, modifier) {
  return tinycolor(color)
    .darken(percentToInt(modifier))
    .toString();
}

export function transparentize(color, modifier) {
  return tinycolor(color)
    .setAlpha(modifier)
    .toString();
}
