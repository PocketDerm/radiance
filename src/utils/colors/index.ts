import tinycolor from 'tinycolor2';

function percentToInt(str: string) {
  // '20%' -> 20
  return parseInt(str.replace(/%/g, ''), 10);
}

type ColorManipulationFunctionType = (
  color: string,
  modifier: string
) => string;

export const lighten: ColorManipulationFunctionType = (color, modifier) =>
  tinycolor(color)
    .lighten(percentToInt(modifier))
    .toString();

export const darken: ColorManipulationFunctionType = (color, modifier) =>
  tinycolor(color)
    .darken(percentToInt(modifier))
    .toString();

export const transparentize = (color: string, modifier: number) =>
  tinycolor(color)
    .setAlpha(modifier)
    .toString();
