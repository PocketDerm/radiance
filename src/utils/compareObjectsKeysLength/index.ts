const compareObjectsKeysLength = (
  object1: Record<string, unknown>,
  object2: Record<string, unknown>,
) => Object.keys(object1).length === Object.keys(object2).length;

export default compareObjectsKeysLength;
