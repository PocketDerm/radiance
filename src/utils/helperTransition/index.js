import React from 'react';
import { Transition } from 'react-transition-group';

import { ANIMATION } from '../../constants';

const getStyleForTransitionState = transitionState => {
  switch (transitionState) {
    case 'entering':
      return {
        opacity: '0',
        maxHeight: '0',
      };
    case 'entered':
      return {
        opacity: '1',
        maxHeight: '50px',
        transitionDuration: ANIMATION.defaultTiming,
        transitionProperty: 'max-height, opacity',
        transitionTimingFunction: 'ease-out',
      };
    case 'exiting':
      return {
        opacity: '0',
        maxHeight: '0',
        transitionDuration: ANIMATION.defaultTiming,
        transitionProperty: 'max-height, opacity',
        transitionTimingFunction: 'ease-in',
      };
    case 'exited':
      return {
        opacity: '0',
        maxHeight: '0',
      };
    default:
      return {};
  }
};

const HelperTransition = ({ children: child, ...props }) => (
  <Transition timeout={350} {...props}>
    {transitionState =>
      React.cloneElement(child, {
        style: getStyleForTransitionState(transitionState),
      })
    }
  </Transition>
);

export default HelperTransition;
