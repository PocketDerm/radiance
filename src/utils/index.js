export * from './colors';
export { default as throwOnUndefinedProperty } from './throwOnUndefinedProperty';
