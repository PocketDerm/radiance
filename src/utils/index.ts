export * from './colors';
export { default as throwOnUndefinedProperty } from './throwOnUndefinedProperty';
export { default as withDeprecationWarning } from './withDeprecationWarning';
export { default as isObject } from './isObject';
export { applyPrimaryThemeVerticalOffset } from './themeStyles';
export * from './injectGlobalStyles/style';
