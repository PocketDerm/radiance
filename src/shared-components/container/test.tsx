import React from 'react';
import renderer from 'react-test-renderer';
import { decorateWithThemeProvider } from 'tests/utils/decorateWithThemeProvider';

import { Container } from './index';

describe('Container UI snapshots', () => {
  const DecoratedContainer = decorateWithThemeProvider(Container);

  test('renders basic container', () => {
    const component = renderer.create(
      <DecoratedContainer>Container Content</DecoratedContainer>,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders container helper components', () => {
    const component = renderer.create(
      <DecoratedContainer>
        <Container.Section>Section 1</Container.Section>
        <Container.Divider />

        <Container.Image src="https://images-na.ssl-images-amazon.com/images/I/61E0pFPOl7L._SL1500_.jpg" />
        <Container.Section>
          <div>Section 2</div>
        </Container.Section>
      </DecoratedContainer>,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders clickable container', () => {
    const component = renderer.create(
      <DecoratedContainer type="clickable">
        Container Content
      </DecoratedContainer>,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders message type container', () => {
    const component = renderer.create(
      <DecoratedContainer type="message">Container Content</DecoratedContainer>,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
