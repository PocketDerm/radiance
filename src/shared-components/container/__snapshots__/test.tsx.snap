// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container UI snapshots renders basic container 1`] = `
.emotion-0 {
  background-color: #FFFFFF;
  border: 1px solid #EDEDF0;
}

<div
  className="emotion-0 emotion-1"
>
  Container Content
</div>
`;

exports[`Container UI snapshots renders clickable container 1`] = `
.emotion-0 {
  background-color: #FFFFFF;
  border: 1px solid #EDEDF0;
  box-shadow: 0px 8px 24px rgba(52,51,82,0.05);
  cursor: pointer;
  -webkit-transition: box-shadow 200ms;
  transition: box-shadow 200ms;
}

.emotion-0:hover {
  box-shadow: 0px 8px 24px rgba(52,51,82,0.10);
}

<div
  className="emotion-0 emotion-1"
  type="clickable"
>
  Container Content
</div>
`;

exports[`Container UI snapshots renders container helper components 1`] = `
.emotion-8 {
  background-color: #FFFFFF;
  border: 1px solid #EDEDF0;
}

.emotion-0 {
  padding: 1.5rem;
}

@media (min-width:992px) {
  .emotion-0 {
    padding-left: 2.5rem;
    padding-right: 2.5rem;
  }
}

.emotion-2 {
  margin: 0 1.5rem;
  border-bottom: 1px solid #EDEDF0;
}

@media (min-width:992px) {
  .emotion-2 {
    margin: 0 2.5rem;
  }
}

.emotion-4 {
  width: 100%;
  overflow: hidden;
  object-fit: cover;
}

<div
  className="emotion-8 emotion-9"
>
  <div
    className="emotion-0 emotion-1"
  >
    Section 1
  </div>
  <div
    className="emotion-2 emotion-3"
  />
  <img
    className="emotion-4 emotion-5"
    src="https://images-na.ssl-images-amazon.com/images/I/61E0pFPOl7L._SL1500_.jpg"
  />
  <div
    className="emotion-0 emotion-1"
  >
    <div>
      Section 2
    </div>
  </div>
</div>
`;

exports[`Container UI snapshots renders message type container 1`] = `
.emotion-0 {
  background-color: #FFFFFF;
  border: 1px solid #EDEDF0;
  border-radius: 16px;
  box-shadow: 0 12px 20px 0 rgba(51,46,84,0.05);
}

<div
  className="emotion-0 emotion-1"
  type="message"
>
  Container Content
</div>
`;
